src/NewMarkownHelpCommand.cs

```diff
@@ -0,0 +1,200 @@
+
+        [Parameter(ParameterSetName = "FromModule")]
+        [Parameter(ParameterSetName = "FromMaml")]
+        public string? HelpVersion { get; set; }
+
+        [Parameter(ParameterSetName = "FromModule")]
+        [Parameter(ParameterSetName = "FromMaml")]
+        public string? Locale { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = "FromMaml")]
+        public string[] MamlFile { get; set; } = Array.Empty<string>();
+
+        [Parameter()]
+        public Hashtable? Metadata { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = "FromModule")]
+        public string[] Module { get; set; } = Array.Empty<string>();
+
+        [Parameter(ParameterSetName = "FromMaml")]
+        public string? ModuleGuid { get; set; }
+
+        [Parameter(ParameterSetName = "FromMaml")]
+        public string? ModuleName { get; set; }
+
+        [Parameter()]
+        public SwitchParameter NoMetadata { get; set; }
+
+        [Parameter(ParameterSetName = "FromCommand")]
+        public string? OnlineVersionUrl { get; set; }
+
+        [Parameter(Mandatory = true)]
+        public string OutputFolder { get; set; } = Environment.CurrentDirectory;
+
+        [Parameter(ParameterSetName = "FromModule")]
+        [Parameter(ParameterSetName = "FromMaml")]
+        public SwitchParameter WithModulePage { get; set; }
+
+        [Parameter(ParameterSetName = "FromMaml")]
+        public SwitchParameter ConvertNotesToList { get; set; }
+
+        [Parameter(ParameterSetName = "FromMaml")]
+        public SwitchParameter ConvertDoubleDashLists { get; set; }
+
+        [Parameter()]
+        public SwitchParameter AlphabeticParamsOrder { get; set; } = true;
+
+        [Parameter()]
+        public SwitchParameter UseFullTypeName { get; set; }
+
+        [Parameter(ParameterSetName = "FromModule")]
+        [Parameter(ParameterSetName = "FromCommand")]
+        public PSSession? Session { get; set; }
+
+        [Parameter(ParameterSetName = "FromModule")]
+        [Parameter(ParameterSetName = "FromMaml")]
+        public string? ModulePagePath { get; set; }
+
+        public SwitchParameter ExcludeDontShow { get; set; }
+
+        #endregion
+
+        protected override void BeginProcessing()
+        {
+            if (Metadata is not null && NoMetadata)
+            {
+                var exception = new InvalidOperationException(Microsoft_PowerShell_PlatyPS_Resources.NoMetadataAndMetadata);
+                ErrorRecord err = new ErrorRecord(exception, "NoMetadataAndMetadata", ErrorCategory.InvalidOperation, Metadata);
+                ThrowTerminatingError(err);
+            }
+        }
+
+        protected override void EndProcessing()
+        {
+            string fullPath = this.SessionState.Path.GetUnresolvedProviderPathFromPSPath(OutputFolder);
+
+            if (!Directory.Exists(fullPath))
+            {
+                Directory.CreateDirectory(fullPath);
+            }
```

Comment:
An edge case: `fullPath` points to an existing file, then `Directory.CreateDirectory` here will fail.

Reply:
Checking if the fullPath is a existing file and throwing a terminating error.

Reply:
Also added a test.